src/security/builtin_plugins/tests/validate_local_identity/src/validate_local_identity_utests.c

```diff
@@ -2141,13 +2265,96 @@ CU_Test(ddssec_builtin_validate_local_identity,with_extended_certificate_check)
                             &candidate_participant_guid,
                             &exception);
 
-
-    /* We expected the validation to have succeeded. */
+    /* We expected the validation to have failed. */
     CU_ASSERT_FATAL (result != DDS_SECURITY_VALIDATION_OK);
     CU_ASSERT (local_identity_handle == DDS_SECURITY_HANDLE_NIL);
 
     dds_security_property_deinit(&participant_qos.property.value);
     reset_exception(&exception);
+}
+
+
+CU_Test(ddssec_builtin_validate_local_identity,crl)
+{
+    DDS_Security_ValidationResult_t result;
+    DDS_Security_IdentityHandle local_identity_handle = DDS_SECURITY_HANDLE_NIL;
+    DDS_Security_GUID_t adjusted_participant_guid;
+    DDS_Security_DomainId domain_id = 0;
+    DDS_Security_Qos participant_qos;
+    DDS_Security_GUID_t candidate_participant_guid;
+    DDS_Security_SecurityException exception = {NULL, 0, 0};
+    DDS_Security_GuidPrefix_t prefix = {0xf0, 0xf1, 0xf2, 0xf3, 0xf4, 0xf5, 0xf6, 0xf7, 0xf8, 0xf9, 0xfa, 0xfb};
+    DDS_Security_EntityId_t entityId = {{0xa0,0xa1,0xa2},0x1};
```

Comment:
So I looked at this one briefly, and I'm not sure I understand where the `0xc1` is supposed to be.  All of the places in this file where `print_guid` is called end up with something like:

```
303: adjusted_participant_guid=f767762b:c1ef319a:70a383fa:a0a1a2:01
303: adjusted_participant_guid=f767762b:c1ef319a:70a383fa:a0a1a2:01
```

Where the `entityKind` member is `0x01`.  That goes both for tests that succeed and ones that fail.  Could you explain a little more about what you are looking for here?

Reply:
You are entirely correct, I simply hadn't realised that all of those tests were using GUIDs with `entityKind` 1. By my reasoning that would make them all "wrong" but it doesn't make any sense to extend this PR to include that matter.

Anyway, in my reading of the DDSI spec, the only possible entity kind for participant GUIDs allowed is `0xc1` , and so I check this for any incoming participant GUID and reject any discovery message for which it doesn't hold:
https://github.com/eclipse-cyclonedds/cyclonedds/blob/5fa204dd61620b24c394ef88d9040944ecbbb470/src/core/ddsi/src/ddsi_plist.c#L1608-L1616

With that in mind, it is quite possible that there is no value whatsoever in having `validate_xyz_identity`, or indeed, any of the security plugins, pay attention to the value of `entityKind` as long as it simply copies the `entityId`/`entityKind` from its input as spec'd. (Which they do here.)

Still, there are some GUIDs sent in, e.g., handshake messages and so I should check if those are all validated properly, or else if a bogus `entityId`/`entityKind` can do no harm. For example, table 52 of the Security spec says:

> The operation shall verify that the first bit of the participant_guid of the ParticipantBuiltinTopic data inside the “c.pdata” is set to 1 and that the following 47 bits match the first 47 bits of the SHA-256 hash of the SubjectName appearing in the IdentityCredential. If this verification fails, the operation shall fail and return ValidationResult_Fail.

According to the spec then, the `entityId`/`entityKind` should be ignored in the validation, and that means I need evidence that there is no harm in a bogus `entityId`/`entityKind`. But all of that is outside the scope of this PR!

Reply:
Thanks for the analysis, makes more sense now.  If you'd like, I can open an issue so someone can look at this later on.  In the meantime, I'll mark this conversation as "Resolved" since there isn't anything to do for this PR.